# HW2 ME581 Final Solution

In [9]:
#Defining functions for various problems
import numpy as np

# Code for infinity norm of a matrix
def inf(a):
    x=(abs(a).sum(axis=1).max())
    return x
# Code for Gauss Elimination with Partial Pivoting
def Gaussp(A):
    n = len(A)
    for i in range(0, n):
        maxtemp = abs(A[i][i])
        pivotrow = i
        
        for k in range(i+1,n):
            if (abs(A[k][i])>maxtemp):
                maxtemp=abs(A[k][i])
                pivotrow=k
                
        for k in range(i,n+1):
            temp=A[i][k]
            A[i][k]=A[pivotrow][k]
            A[pivotrow][k]=temp
            
        for k in range(i+1,n):
            c=-(A[k][i])/(A[i][i])
           # if (c<1e-10):
            #    c=0
            for j in range(i, n+1):
                if i == j:
                    A[k][j] = 0
                else:
                    A[k][j] += c * A[i][j]

        
    x = [0 for i in range(n)]
    for i in range(n-1, -1, -1):
        x[i] = A[i][n]/A[i][i]
        for k in range(i-1, -1, -1):
            A[k][n] -= A[k][i] * x[i]
    return x

#Code for Gauss Elimination with scaled pivoting
def Gscale(v):
    n=len(v)
    row=np.empty([n,1])
    s=(np.abs(v)[:,:-1]).max(axis=1)
    
    for i in range(0,n):
        for k in range(i,n):
            row[k]=v[k][i]/s[k]
            temp=abs(row[i])
            srow=i
        for k in range(i,n):
            if (abs(row[k])>temp):
                temp=abs(row[k])
                srow=k
        for k in range(i,n+1):
            temp=v[i][k]
            v[i][k]=v[srow][k]
            v[srow][k]=temp
        for k in range(i+1,n):
            c=-v[k][i]/v[i][i]
            #if(c<(1e-10)):
             #   c=0
            for j in range(i,n+1):
                v[k][j]=v[k][j]+(c*v[i][j])
    y = [0 for i in range(n)]
    for i in range(n-1, -1, -1):
        y[i] = v[i][n]/v[i][i]
        for k in range(i-1, -1, -1):
            v[k][n] -= v[k][i] * y[i]
    return y

## Problem 1

A system of equations Ax=b is given as

$\left[\begin{array}{cc}2.01 & 1.99 \\1.99 & 2.01  \\ \end{array} \right]$$x$=$\left[\begin{array}{c}4\\4\\\end{array} \right]$

The solution to this system of equations is
$x=\left[ \begin{array}{c}1\\1\\\end{array} \right]$  ,
and one approximate solution is
$\widetilde{x}=\left[ \begin{array}{c}2\\0\\ \end{array} \right]$ .<br>
1) Compute the error $e=x-\widetilde{x}$

2) Compute the residual $r=A\widetilde{x}-b$.

3) Using the ${l}_\infty$ norm, compute the relative error $\frac{{||e||}_\infty}{{||x||}_\infty}$.

4)	Using the ${l}_\infty$, compute the condition number ${k}_\infty$.

5)	Using the ${l}_\infty$, compute the relative residual $\frac{{||r||}_\infty}{{||b||}_\infty}$.

6)	Compute the product of the condition number and the relative residual.

7)	Compare the relative error to the product of the condition number and the relative residual.

### Solution

In [10]:
x1=np.matrix([1,1])
x2=np.matrix([2,0])
A1=np.matrix([[2.01,1.99],[1.99,2.01]])
B1=np.matrix([4,4])

#1)
e=x1-x2
print("1) The error e = ",e)

#2)
r=(A1*x2.T)-B1
print("\n2) The residual r = \n",r)

#3)
r2=inf(e)/inf(x1)
print("\n3) The e/x ratio using the l infinity norm is ",r2)

#4)
k=inf(A1)*inf(A1.I)
print("\n4) The condition number k =",k)

#5)
r3=inf(r)/inf(B1)
print("\n5) The relative residual r/b using the l infinity norm is ",r3)

#6)
print("\n6) The product of the condition number and the relative residual is ",k*r3)




1) The error e =  [[-1  1]]

2) The residual r = 
 [[ 0.02  0.02]
 [-0.02 -0.02]]

3) The e/x ratio using the l infinity norm is  1.0

4) The condition number k = 200.0

5) The relative residual r/b using the l infinity norm is  0.005

6) The product of the condition number and the relative residual is  1.0


7) The relative error is seen to be equal than the product of the condition number and the relative residual Thus it satisfies the condition:
$\dfrac{{||e||}}{{||x||}} \leqslant ||{A}||*||{{A}^{-1}}||*\dfrac{{||r||}}{{||b||}}$ 

## Problem 2
Let $\qquad$A=$\left[ \begin{array}{cccc}5.1 & 8.7 \\2.4 & 4.1  \\ \end{array} \right]\qquad$ B=$\left[ \begin{array}{c}9.48\\4.48\\\end{array} \right]$<br><br>
1)	Using the ${l}_\infty$ norm, compute the condition number ${k}_\infty(A)$.<br>

2)	Solve the system of equations $Ax=b$ for $x$.<br>

3)	Perturb the coefficient matrix A and the right-side vector b by
$δA=\left[ \begin{array}{rr}-0.001 & 0 \\0.001 & 0  \\ \end{array} \right]$   and   $δb=\left[ \begin{array}{cc}0.05\\-0.05\\\end{array} \right]$
and solve the resulting perturbed system of equations $(A+δA)\widetilde{x}=(b+δb)$ for the approximate solution $\widetilde{x}$.<br>

4)	Using the ${l}_\infty$ norm, compute the actual value of the relative change in the solution,$\dfrac{{||δx||}_\infty}{||x||_\infty}$ for the perturbation in part (3).<br>

5)	Using the ${l}_\infty$ norm, compute the theoretical upper bound of the relative change in the solution,$\dfrac{{||δx||}_\infty}{||x||_\infty}$ for the perturbation in part (3).<br>

6)	For the perturbation in part (3), compare the actual value of $\dfrac{{||δx||}_\infty}{||x||_\infty}$  to its theoretical upper bound.<br>

7)  Perturb the coefficient matrix A and the right-side vector b by
$δA=\left[ \begin{array}{rrrr}0.001 & -0.001 \\-0.001 & 0.001  \\ \end{array} \right]$   and   $δb=\left[ \begin{array}{cccc}-0.1\\0.1\\\end{array} \right]$
and solve the resulting perturbed system of equations $(A+δA)\widetilde{x}=(b+δb)$ for the approximate solution $\widetilde{x}$.<br>

8) Using the ${l}_\infty$ norm, compute the actual value of the relative change in the solution, $\dfrac{{||δx||}_\infty}{||x||_\infty}$ for the perturbation in part (7).<br>

9)	Using the ${l}_\infty$ norm, compute the theoretical upper bound of the relative change in the solution,$\dfrac{{||δx||}_\infty}{||x||_\infty}$ for the perturbation in part (7).<br>

10)	For the perturbation in part (7), compare the actual value of $\dfrac{{||δx||}_\infty}{||x||_\infty}$  to its theoretical upper bound.<br>

### Solution

In [11]:
A3=np.matrix([[5.1,8.7],[2.4,4.1]])
B3=np.matrix([[9.48],[4.48]])
dA1=np.matrix([[-0.001,0],[0.001,0]])
dB1=np.matrix([[0.05],[-0.05]])
dA2=np.matrix([[0.001,-0.001],[-0.001,0.001]])
dB2=np.matrix([[-0.1],[0.1]])

#1)
m1=inf(A3)*inf(A3.I)
print("\n1) The condition number k is ",m1)

#2)
ans=(A3.I)*(B3)
print("\n2) Solution for the Ax=b is given by x=\n",ans)

#3)
ans1=((A3+dA1).I)*(B3+dB1)
print("\n3) Solution for the first perturbed system is given by x1=\n",ans1)

#4)
ans2=inf(ans1-ans)/inf(ans)
print("\n4) The relative change in solution using the infinity norm is ",ans2)

#5)
up= (m1*((inf(dB1)/inf(B3))+(inf(dA1)/inf(A3))))/(1-(m1*(inf(dA1)/inf(A3))))
print("\n5) The theoretical upper bound for the first perturbed system is ",up)



1) The condition number k is  5888.0

2) Solution for the Ax=b is given by x=
 [[-3.6]
 [ 3.2]]

3) Solution for the first perturbed system is given by x1=
 [[ 30.93023256]
 [-17.03255814]]

4) The relative change in solution using the infinity norm is  9.59173126615

5) The theoretical upper bound for the first perturbed system is  54.9096261407


6) 	The relative change in the solution is seen to be less than its theoretical upper bound:<br><br>
$(\dfrac{{||δx||}_\infty}{||x||_\infty}=9.59)< 54.91$<br><br>

In [12]:
#7)
ans3=((A3+dA2).I)*(B3+dB2)
print("\n7) Solution for the second perturbed system is given by x2=\n",ans3)

#8)
ans4=inf(ans3-ans)/inf(ans)
print("\n8) The relative change in solution using the infinity norm is ",ans4)

#9)
up1= (m1*((inf(dB2)/inf(B3))+(inf(dA2)/inf(A3))))/(1-(m1*(inf(dA2)/inf(A3))))
print("\n9) The theoretical upper bound for the second perturbed system is ",up1)


7) Solution for the second perturbed system is given by x2=
 [[-27.31689861]
 [ 17.09662028]]

8) The relative change in solution using the infinity norm is  6.58802739121

9) The theoretical upper bound for the second perturbed system is  429.293440736


10) The relative change in the solution is seen to be less than its theoretical upper bound:<br><br>
$(\dfrac{{||δx||}_\infty}{||x||_\infty}=6.58)< 429.3$<br><br>

## Problem 3
Solve the augmented matrix $\qquad$A =$\qquad\left[\begin{array}{rrrrr}3.0 & 1.0 & 4 & -1 &  7 \\2.0 & -2.0 & -1.0 & 2.0 &  1\\5.0 & 7.0 & 14.0 & -9.0 &  21.0\\1.0 & 3.0 & 2.0 & 4.0 &  -4.0\\\end{array}\right]$<br>

(i)By Gaussian Elimination with Partial Pivoting.<br>

(ii)By Gaussian Elimination with Scaled Partial Pivoting


### Solution

In [13]:
#(i)
A5=np.array([[3.0,1.0,4.0,-1.0,7.0],
             [2.0,-2.0,-1.0,2.0,1.0],
             [5.0,7.0,14.0,-9.0,21.0],
             [1.0,3.0,2.0,4.0,-4.0]])
xans=Gaussp(A5)
print("\n(i) The solution with Gauss Elimination with Partial Pivoting is given by:")
print(xans)

#(ii)
A5=np.array([[3.0,1.0,4.0,-1.0,7.0],
             [2.0,-2.0,-1.0,2.0,1.0],
             [5.0,7.0,14.0,-9.0,21.0],
             [1.0,3.0,2.0,4.0,-4.0]])
xans2=Gscale(A5)
print("\n(ii) The solution with Guass Elimination with Scaled Pivoting is given by:")
print(xans2)


(i) The solution with Gauss Elimination with Partial Pivoting is given by:
[0.99999999999999822, -1.0000000000000016, 1.0000000000000018, -0.99999999999999933]

(ii) The solution with Guass Elimination with Scaled Pivoting is given by:
[1.0, -1.0, 1.0, -1.0]


## Problem 4
$\qquad$A =$\qquad\left[\begin{array}{rrrrrr}-9 &-11 & -21 & 63 & -252 & -356 \\70 & -69 & 141 & -421 & 1684 & 2385 \\-575 & 575 & -1149 & 3451 & -13801 &-19551\\3891 & -3891 & 7782 & -23345 & 93365 & 132274\\1024 & -1024 & 2048 &-6144 & 24572& 34812  \\ \end{array} \right]\qquad$<br><br>

(i) Solve the augmented matrix by means of Gaussian Elimination with Partial Pivoting in double precision.<br>

(ii) Using the ${l}_\infty$, estimate the condition number of the coefficient matrix based on your result. The exact solution for this problem is given by 
$x=\left[\begin{array}{ccccc}1 & -1 &1 &-1 & 1\\\end{array}\right]$

### Solution

In [14]:
#(i)
A6=np.array([[-9.,11.,-21.,63.,-252.,-356.],
             [70.,-69.,141.,-421.,1684.,2385.],
             [-575.,575.,-1149.,3451.,-13801.,-19551.],
             [3891.,-3891.,7782.,-23345.,93365.,132274.],
             [1024.,-1024.,2048.,-6144.,24572.,34812.]])
xans3=Gaussp(A6)
xans4=np.reshape(xans3,(5,1))
print("\n(i) The Solution by means of Gauss Elimination by Partial pivoting is given by x=")
for i in range(0,5):
    print(xans3[i])

A7=np.matrix([[-9.,11.,-21.,63.,-252.],
             [70.,-69.,141.,-421.,1684.],
             [-575.,575.,-1149.,3451.,-13801.],
             [3891.,-3891.,7782.,-23345.,93365.],
             [1024.,-1024.,2048.,-6144.,24572.]])
k3= inf(A7)*inf(A7.I)
print("(ii) The condition number for the given system is given by k=",k3)


(i) The Solution by means of Gauss Elimination by Partial pivoting is given by x=
1.0
-0.864684385603
0.0850087026453
5.2696234671
2.64956177552
(ii) The condition number for the given system is given by k= 4.4062629868e+17


(ii)(Alternate)For ${k}_\infty = \approx {10}^{r}$ we expect to lose r decimal digits of precision. Our solution in double precision is inaccurate by the first digit, meaning that it has lost all of the 15 to 17 digits of double precision. Thus,
$[{k}_\infty]$$\approx$${10}^{r}$$\approx$${10}^{15}-{10}^{17}$


## Problem 5
Determine the member and reaction forces within the plane truss shown in Figure 1 when the truss is subjected to each of the following loading configurations.<br>

(a)	500-pound forces directed vertically downward at nodes#3 and #5, and a 1000-pound force directed vertically downward at node#4.<br>

(b)	A 500-pound force directed vertically downward at nodes#3, a 1000-pound force directed vertically downward at node#4, a 1500-pound force directed vertically downward at node#5.<br>

(c)	A 1500-pound force directed vertically downward at nodes#3, a 1000-pound force directed vertically downward at node#4, a 500-pound force directed vertically downward at node#5.<br>

(d)	500-pound force acting at node#4, and a 1000-pound force acting at node #3, both forces acting horizontally to the right.<br>

(e)	500-pound force acting at node#4, and a 1000-pound force acting at node #5, both forces acting horizontally to the left.<br>


Solve the problem using your GE code with partial pivoting. Show the augmented matrix and the resulting forces for each case.<br>
![alt text](diagram.png "Title")

### Solution


Now the equations when th entire system can be given as.<br>

#### For node1<br>

$-F1*cos\theta - F3*cos\beta + Fh=0\qquad$-(1)<br> 

$-F1*sin\theta - F3*sin\beta + Fv=0\qquad$-(2)<br> 

#### For node2<br>

$F8*Cos\alpha - F9*cos\theta - F2*cos\alpha + F1*cos\theta = 0\qquad$-(3)<br>

$F8*sin\alpha + F2*sin\alpha + F5 + F1*sin\theta + F9*sin\theta =0\qquad$-(4)<br>

#### For node3<br>

$F2*cos\alpha + F4*cos\gamma - F3*cos\beta + AF_{3x} =0\qquad$-(5)<br>

$-F4*sin\gamma - F2*sin\alpha + F3*sin\beta + AF_{3y} =0\qquad$-(6)<br>

#### For node4<br>

$F6*cos\gamma - F4*cos\gamma + AF_{4x} =0\qquad$-(7)<br>

$-F5 + F6*sin\gamma + F4*sin\gamma + AF_{4y} =0\qquad$-(8)<br>

#### For node5<br>

$-F7*cos\beta + F6*cos\gamma - F8*cos\alpha + AF_{5x} =0\qquad$-(9)<br>

$F6*sin\gamma - F7*sin\beta - F8*sin\alpha + AF_{5y} =0\qquad$-(10)<br>

#### For node6<br>

$F9*cos\theta + F7*cos\beta =0\qquad$-(11)<br>

$-F9*sin\theta - F7*sin\beta + Fr=0\qquad$-(12)<br>

Thus it can be seen that we have 12 variables and 12 equations<br><br>

Here AF is the externally applied force. The subscript denotes the node number and the direction.
$\alpha$ is the angle made by F2 with the x axis.$\beta$ is the angle made by F3 with the x axis. $\gamma$ is the angle made by F4 with the x axis and $\theta$ is the angle made by F1 with the axis.

## (i) 


In [15]:
AF3Y=-500
AF4Y=-1000
AF5Y=-500
AF3X=0
AF4X=0
AF5X=0
c4=30/(np.power(900+225,0.5))
s4=15/(np.power(900+225,0.5))
c2=15/(np.power(400+225,0.5))
s2=20/(np.power(400+225,0.5))
c3=15/(np.power(25+225,0.5))
s3=5/(np.power(25+225,0.5))
c1=c3
s1=s3
A9=np.array([[-c4,0.,-c2,0.,0.,0.,0.,0.,0.,0.,0.,1,0.],
             [c4,-c1,0.,0.,0.,0.,0.,c1,-c4,0.,0.,0.,0.],
             [0.,c1,c2,-c3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3X],
             [0.,0.,0.,c3,0.,-c3,0.,0.,0.,0.,0.,0.,0.,-AF4X],
             [0.,0.,0.,0.,0.,c3,-c2,-c1,0.,0.,0.,0.,-AF5X],
             [0.,0.,0.,0.,0.,0.,c2,0.,c4,0.,0.,0.,0.],
             [-s4,0.,-s2,0.,0.,0.,0.,0.,0.,0.,1.,0.,0.],
             [s4,s1,0.,0.,1.,0.,0.,s1,s4,0.,0.,0.,0.],
             [0.,-s1,s2,-s3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3Y],
             [0.,0.,0.,s3,-1,s3,0.,0.,0.,0.,0.,0.,-AF4Y],
             [0.,0.,0.,0.,0.,-s3,s2,-s1,0.,0.,0.,0.,-AF5Y],
             [0.,0.,0.,0.,0.,0.,-s2,0.,-s4,1.,0.,0.,0.]])
#a)

x9=Gaussp(A9)
force=['F1','F2','F3','F4','F5','F6','F7','F8','F9','Fr','Fv','Fh']


#b)
AF3Y=-500
AF4Y=-1000
AF5Y=-1500
AF3X=0
AF4X=0
AF5X=0
A10=np.array([[-c4,0.,-c2,0.,0.,0.,0.,0.,0.,0.,0.,1,0.],
             [c4,-c1,0.,0.,0.,0.,0.,c1,-c4,0.,0.,0.,0.],
             [0.,c1,c2,-c3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3X],
             [0.,0.,0.,c3,0.,-c3,0.,0.,0.,0.,0.,0.,0.,-AF4X],
             [0.,0.,0.,0.,0.,c3,-c2,-c1,0.,0.,0.,0.,-AF5X],
             [0.,0.,0.,0.,0.,0.,c2,0.,c4,0.,0.,0.,0.],
             [-s4,0.,-s2,0.,0.,0.,0.,0.,0.,0.,1.,0.,0.],
             [s4,s1,0.,0.,1.,0.,0.,s1,s4,0.,0.,0.,0.],
             [0.,-s1,s2,-s3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3Y],
             [0.,0.,0.,s3,-1,s3,0.,0.,0.,0.,0.,0.,-AF4Y],
             [0.,0.,0.,0.,0.,-s3,s2,-s1,0.,0.,0.,0.,-AF5Y],
             [0.,0.,0.,0.,0.,0.,-s2,0.,-s4,1.,0.,0.,0.]])
x10=Gaussp(A10)

#c)
AF3Y=-1500
AF4Y=-1000
AF5Y=-500
AF3X=0
AF4X=0
AF5X=0
A11=np.array([[-c4,0.,-c2,0.,0.,0.,0.,0.,0.,0.,0.,1,0.],
             [c4,-c1,0.,0.,0.,0.,0.,c1,-c4,0.,0.,0.,0.],
             [0.,c1,c2,-c3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3X],
             [0.,0.,0.,c3,0.,-c3,0.,0.,0.,0.,0.,0.,0.,-AF4X],
             [0.,0.,0.,0.,0.,c3,-c2,-c1,0.,0.,0.,0.,-AF5X],
             [0.,0.,0.,0.,0.,0.,c2,0.,c4,0.,0.,0.,0.],
             [-s4,0.,-s2,0.,0.,0.,0.,0.,0.,0.,1.,0.,0.],
             [s4,s1,0.,0.,1.,0.,0.,s1,s4,0.,0.,0.,0.],
             [0.,-s1,s2,-s3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3Y],
             [0.,0.,0.,s3,-1,s3,0.,0.,0.,0.,0.,0.,-AF4Y],
             [0.,0.,0.,0.,0.,-s3,s2,-s1,0.,0.,0.,0.,-AF5Y],
             [0.,0.,0.,0.,0.,0.,-s2,0.,-s4,1.,0.,0.,0.]])
x11=Gaussp(A11)

#d)
AF3Y=0
AF4Y=0
AF5Y=0
AF3X=1000.0
AF4X=500.0
AF5X=0
A12=np.array([[-c4,0.,-c2,0.,0.,0.,0.,0.,0.,0.,0.,1.,0.],
             [c4,-c1,0.,0.,0.,0.,0.,c1,-c4,0.,0.,0.,0.],
             [0.,c1,c2,-c3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3X],
             [0.,0.,0.,c3,0.,-c3,0.,0.,0.,0.,0.,0.,-AF4X],
             [0.,0.,0.,0.,0.,c3,-c2,-c1,0.,0.,0.,0.,-AF5X],
             [0.,0.,0.,0.,0.,0.,c2,0.,c4,0.,0.,0.,0.],
             [-s4,0.,-s2,0.,0.,0.,0.,0.,0.,0.,1.,0.,0.],
             [s4,s1,0.,0.,1.,0.,0.,s1,s4,0.,0.,0.,0.],
             [0.,-s1,s2,-s3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3Y],
             [0.,0.,0.,s3,-1.0,s3,0.,0.,0.,0.,0.,0.,-AF4Y],
             [0.,0.,0.,0.,0.,-s3,s2,-s1,0.,0.,0.,0.,-AF5Y],
             [0.,0.,0.,0.,0.,0.,-s2,0.,-s4,1.,0.,0.,0.]])
x12=Gaussp(A12)


#e)
AF3Y=0
AF4Y=0
AF5Y=0
AF3X=0
AF4X=-500.0
AF5X=-1000.0
A13=np.array([[-c4,0.,-c2,0.,0.,0.,0.,0.,0.,0.,0.,1.,0.],
             [c4,-c1,0.,0.,0.,0.,0.,c1,-c4,0.,0.,0.,0.],
             [0.,c1,c2,-c3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3X],
             [0.,0.,0.,c3,0.,-c3,0.,0.,0.,0.,0.,0.,-AF4X],
             [0.,0.,0.,0.,0.,c3,-c2,-c1,0.,0.,0.,0.,-AF5X],
             [0.,0.,0.,0.,0.,0.,c2,0.,c4,0.,0.,0.,0.],
             [-s4,0.,-s2,0.,0.,0.,0.,0.,0.,0.,1.,0.,0.],
             [s4,s1,0.,0.,1.,0.,0.,s1,s4,0.,0.,0.,0.],
             [0.,-s1,s2,-s3,0.,0.,0.,0.,0.,0.,0.,0.,-AF3Y],
             [0.,0.,0.,s3,-1.0,s3,0.,0.,0.,0.,0.,0.,-AF4Y],
             [0.,0.,0.,0.,0.,-s3,s2,-s1,0.,0.,0.,0.,-AF5Y],
             [0.,0.,0.,0.,0.,0.,-s2,0.,-s4,1.,0.,0.,0.]])
x13=Gaussp(A13)
print("Case A")
print("\t\t\tA")
for i in range(0,12):
    print(force[i],"\t\t",x9[i])
print("\nCase B")
print("\t\t\tB")
for i in range(0,12):
    print(force[i],"\t\t",x10[i])
print("\nCase C")
print("\t\t\tC")
for i in range(0,12):
    print(force[i],"\t\t",x11[i])
print("\nCase D")
print("\t\t\tD")
for i in range(0,12):
    print(force[i],"\t\t",x12[i])
print("\nCase E")
print("\t\t\tE")
for i in range(0,12):
    print(force[i],"\t\t",x13[i])

Case A
			A
F1 		 -1341.6407865
F2 		 1106.79718106
F3 		 2000.0
F4 		 2371.70824513
F5 		 500.0
F6 		 2371.70824513
F7 		 2000.0
F8 		 1106.79718106
F9 		 -1341.6407865
Fr 		 1000.0
Fv 		 1000.0
Fh 		 0.0

Case B
			B
F1 		 -1677.05098312
F2 		 1581.13883008
F3 		 2500.0
F4 		 3162.27766017
F5 		 1000.0
F6 		 3162.27766017
F7 		 3500.0
F8 		 948.683298051
F9 		 -2347.87137637
Fr 		 1750.0
Fv 		 1250.0
Fh 		 0.0

Case C
			C
F1 		 -2347.87137637
F2 		 948.683298051
F3 		 3500.0
F4 		 3162.27766017
F5 		 1000.0
F6 		 3162.27766017
F7 		 2500.0
F8 		 1581.13883008
F9 		 -1677.05098312
Fr 		 1250.0
Fv 		 1750.0
Fh 		 0.0

Case D
			D
F1 		 -1956.55948031
F2 		 -131.761569174
F3 		 416.666666667
F4 		 1185.85412256
F5 		 916.666666667
F6 		 1712.90039926
F7 		 1083.33333333
F8 		 1027.74023955
F9 		 -726.722092687
Fr 		 541.666666667
Fv 		 -541.666666667
Fh 		 -1500.0

Case E
			E
F1 		 1956.55948031
F2 		 -395.284707521
F3 		 -416.666666667
F4 		 -658.807845868
F5 		 -583.333333333
F6 		 